sdk/ml/azure-ai-ml/azure/ai/ml/operations/_compute_operations.py

```diff
@@ -39,11 +42,13 @@ def __init__(
         operation_scope: OperationScope,
         operation_config: OperationConfig,
         service_client: ServiceClient022023Preview,
+        service_client_2024: ServiceClient042024Preview,
         **kwargs: Dict,
     ) -> None:
         super(ComputeOperations, self).__init__(operation_scope, operation_config)
         ops_logger.update_info(kwargs)
         self._operation = service_client.compute
+        self._operation2024 = service_client_2024.compute
```

Comment:
why not change all of compute to be on the new API version? Anything that is GA in a previous version will be GA'd in the new preview version, so this should be a safe upgrade.

Reply:
I do see this pattern of different version clients used, I am not comfortable making the changes to other API close to release cycle, I am adding a work item and resolving your comment as this point of time.
[Task 3372541](https://msdata.visualstudio.com/Vienna/_workitems/edit/3372541): Consolidate the rest client version for Compute operations.